Icon Sizes (#2116)
MicrosoftDocs/edge-developer: microsoft-edge/progressive-web-apps-chromium/how-to/icon-theme-color.md

Learn more about [the icons member](https://developer.mozilla.org/docs/Web/Manifest/icons).

> ## Icon Image Sizes

Reviewer: No need for capital letters on each word. So, this should be `Icon image sizes`. See the writing guidelines about capitalization here: https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=33685

Author: fixed in the edit.

Reviewer: Thanks. I forgot to mention earlier that the 3 other headers also have to be changed in the same way: "Image Level Summary" and "Image Descriptions" and "Target Sized Images".

Author: fixed.

> ## Icon Image Sizes

Reviewer: Please add a html comment separator before this H2 heading. Like we do for all other H2 headings in this repository.

Author: fixed

Your PWA can be enhanced on Windows by supplying the images with specific dimensions in your web app manifest. We recommend choosing one of the options below:

> * Level 1: (Basic image support: 512x512) This is the base image from which to generate missing images

Reviewer: Missing period at the end of this sentence.

Author: fixed

> * Level 2: (Titles) At this level, your web app manifest contains tile images for the default (1x) display scale. Each of the images must be in PNG format and purpose any.Here is a list of recommended images and sizes.

Reviewer: ```suggestion
* Level 2: (Titles) At this level, your web app manifest contains tile images for the default (1x) display scale. Each of the images must be in PNG format and have their `purpose` property set to `any`. Here is a list of recommended images and sizes.
```

    * 44x44 - app icon
    * 71x71 - small tile
    * 150x150 - medium tile
    * 350x150 - wide tile
    * 310x310 - large tile
    * 50x50 - store logo
    * 620x300 - splash screen

> * Level 3: (Tiles with display scales) At this level, your web app manifest icons should include images with these dimensions:

Reviewer: I suggest rephrasing to make this a bit easier to understand, by using some of your content that's further down:

```suggestion
* Level 3: (Tiles with display scales) At this level, your web app manifest should contain tile images for all Windows display scale sizes. Display scale is a user-configuration in Windows (users can change it by going to **Settings** > **Display** > **Scale**). Your web app manifest icons should include images with these dimensions:
```

Reviewer: This was not addressed. I think we should do it because I find the current structure confusing. Currently, we give a bit of a description here, then a long list of icon sizes, and finally at the bottom of the list, some more description.

I suggest applying my suggested change in the previous comment, and then deleting the description at the bottom of the list.

This way, people understand upfront what level 3 is about, and can then decide to go through the list if they want.

    * 188x188 - medium tile 1.25x display scale
    * 225x225 - medium tile 1.5x display scale
    * 300x300 - medium tile 2x display scale
    * 600x600 - medium tile 4x display scale
    * 350x150 - wide tile
    * 388x188 - wide tile 1.25x display scale
    * 465x225 - wide tile 1.5x display scale
    * 620x300 - wide tile 2x display scale
    * 1240x600 - wide tile 4x display scale
    * 310x310 - large tile
    * 388x388 - large tile 1.25x display scale
    * 465x465 - large tile 1.5x display scale
    * 620x620 - large tile 2x display scale
    * 1240x1240 - large tile 4x display scale
    * 50x50 - store tile
    * 63x63 - store tile 1.25x display scale
    * 75x75 - store tile 1.5x display scale
    * 100x100 - store tile 2x display scale
    * 200x200 - store tile 4x display scale
    * 620x300 - splash screen
    * 775x375 - splash screen 1.25x display scale  
    * 930x450 - splash screen 1.5x display scale
    * 1240x600 - splash screen 2x display scale
    * 2480x1200 - splash screen 4x display scale

For level 3, your web app manifest should contain tile images for all Windows display scale sizes.

Display scales are user-configurable in Windows:

> ![Display Sample](../media/windows-image-display-scales.png) 

Reviewer: And then you can remove this part since it's now above. Also, I don't think this image is helping much here. Just saying that the scale is configurable is enough, this article doesn't need to explain where to change it.

Author: removed Display scales are user-configurable in Windows:
![Display Sample](../media/windows-image-display-scales.png)

> * Level 4 (Tiles, display scales, and target sizes)
> Your web app manifest icons should include images with these dimensions:

Reviewer: Same comment here: let's use the content that you have at the bottom of this item here instead. It's valuable and should probably be read first, before the list of sizes. Here's a suggestion:

```suggestion
* Level 4 (Tiles, display scales, and target sizes) At this level you supply images for tiles with display scales and target size images for display in various surfaces in Windows, including taskbar, start menu, task manager, ALT+Tab task switcher, and more. This provides the best experience for your users, but also requires the most developer effort. Your web app manifest icons should include images with these dimensions:
```

    * 16x16 - target size for taskbar, start menu, task manager
    * 20x20 - target size for taskbar, start menu, task manager
    * 24x24 - target size for taskbar, start menu, task manager
    * 30x30 - target size for taskbar, start menu, task manager
    * 32x32 - target size for taskbar, start menu, task manager
    * 36x36 - target size for taskbar, start menu, task manager
    * 40x40 - target size for taskbar, start menu, task manager
    * 44x44 - target size for taskbar, start menu, task manager
    * 48x48 - target size for taskbar, start menu, task manager
    * 60x60 - target size for taskbar, start menu, task manager
    * 64x64 - target size for taskbar, start menu, task manager
    * 72x72 - target size for taskbar, start menu, task manager
    * 80x80 - target size for taskbar, start menu, task manager
    * 96x96 - target size for taskbar, start menu, task manager
    * 256x256 - target size for taskbar, start menu, task manager
In level 4, you supply images for tiles with display scales and target size images for display in various surfaces in Windows, including taskbar, start menu, task manager, ALT+Tab task switcher, and more.

> This provides the best experience for your users, but also requires the most developer effort.

Reviewer: You can then remove this part since it's now above.

Author: removed

### Image Level Summary

Consider, for example, how your app icon might show different content depending on the available surface area:

> * Small surface: App icon (44x44) in taskbar, showing a small, abbreviated logo: ![The small size icon](../media/windows-image-app-icon.png) 

Reviewer: We typically use inline images only for small product icons that are short enough to nicely fit on a line of text.
This is not the case for this one, so please can you move the image on its own line:

```suggestion
* Small surface: App icon (44x44) in the taskbar, showing a small, abbreviated logo:

   ![The small size icon in the taskbar](../media/windows-image-app-icon.png) 
```

Same comment for the other 2 bullet points below.

Author: done

* Larger surface: Medium tile (150x150) in start menu, showing the full logo of the app:  ![The medium size icon.](../media/windows-image-small-tile.png) 

* Wide surface: Wide tile in start menu, showing a more articulated app icon: ![The wide size icon](../media/windows-image-wide-tile.png) 

Notice how the content of the icon changed. This is possible only through the use of higher levels of icon support. We recommend developers choose higher levels to give the best experience for your Windows users.  

### Icon Descriptions

Below you’ll find a description of each app icon and where they show up in Windows.

![App Icon](../media/windows-image-app-icon.png) 

> The normal app icon that's shown in the start menu, task bar, or task manager.

Reviewer: Could you please move the image below the text? It seems more logical to me to first read the icon description, and then see the image example below that.

Also, please add 1 empty line between the text and the image, in order to make sure the image is on its own line, and not next to the text.

Same comment for the other sections below.

Author: done!

* 50x50
* 63x63 (1.25x scale)
* 75x75 (1.5x scale)
* 100x100 (2x scale)
* 200x200 (4x scale)

![Splash Screen](../media/windows-image-splash.png)
The splash screen asset is shown as the splash screen for your app. Currently supported only in classic package. In the future, we may add support for the modern hosted app package as well.

* 620x300
* 775x375 (1.25x scale)
* 930x450 (1.5x scale)
* 1240x600 (2x scale)
* 2480x1200 (4x scale)

### Target Sized Images

In addition to the standard scale factor sizes described above, we also recommend creating “target-size” assets. We call these assets target-size because they target specific sizes, such as 16 pixels, rather than specific scale factors, such as 400. Target-size assets are for Windows surfaces that don’t use the scaling plateau system.

![Shown in start jump list, shortcuts, control panel:](../media/windows-image-target-size.png)

* 16x16 (recommended)
* 20x20
* 24x24 (recommended)
* 30x30
* 32x32 (recommended)
* 36x36
* 40x40
* 48x48 (recommended)
* 60x60
* 64x64
* 72x72
* 80x80
* 96x96
* 256x256 (recommended)

> <!-- ====================================================================== -->

Reviewer: Please leave this html comment separator here. We have a convention on this repo where we add this separate before each H2 heading. This makes it easier to read the markdown code in VS Code.

Author: added before <!-- ====================================================================== -->
## Icon image sizes

Learn more about [the icons member](https://developer.mozilla.org/docs/Web/Manifest/icons).

<!-- ====================================================================== -->
## Icon image sizes

Your PWA can be enhanced on Windows by supplying the images with specific dimensions in your web app manifest. We recommend choosing one of the options below:

* Level 1: (Basic image support: 512x512) This is the base image from which to generate missing images.

> * Level 2: (Titles) At this level, your web app manifest contains tile images for the default (1x) display scale. Each of the images must be in PNG format and purpose any.Here is a list of recommended images and sizes.

Reviewer: ```suggestion
* Level 2: (Tiles) At this level, your web app manifest contains tile images for the default (1x) display scale. Each of the images must be in PNG format and have the `any` purpose set. Here is a list of recommended images and sizes.
```

    * 107x107 - small tile 1.5x display scale
    * 142x142 - small tile 2x display scale
    * 284x284 - small tile 4x display scale
    * 150x150 - medium tile
    * 188x188 - medium tile 1.25x display scale
    * 225x225 - medium tile 1.5x display scale
    * 300x300 - medium tile 2x display scale
    * 600x600 - medium tile 4x display scale
    * 350x150 - wide tile
    * 388x188 - wide tile 1.25x display scale
    * 465x225 - wide tile 1.5x display scale
    * 620x300 - wide tile 2x display scale
    * 1240x600 - wide tile 4x display scale
    * 310x310 - large tile
    * 388x388 - large tile 1.25x display scale
    * 465x465 - large tile 1.5x display scale
    * 620x620 - large tile 2x display scale
    * 1240x1240 - large tile 4x display scale
    * 50x50 - store tile
    * 63x63 - store tile 1.25x display scale
    * 75x75 - store tile 1.5x display scale
    * 100x100 - store tile 2x display scale
    * 200x200 - store tile 4x display scale
    * 620x300 - splash screen
    * 775x375 - splash screen 1.25x display scale  
    * 930x450 - splash screen 1.5x display scale
    * 1240x600 - splash screen 2x display scale
    * 2480x1200 - splash screen 4x display scale

> For level 3, your web app manifest should contain tile images for all Windows display scale sizes.

Reviewer: As mentioned in the previous comment, this can then be deleted.

Reviewer: Can you please delete this sentence? It is already present earlier in the article, where the Level 3 images are described (line 77).

Author: done

    * 620x300 - splash screen
    * 775x375 - splash screen 1.25x display scale
    * 930x450 - splash screen 1.5x display scale
    * 1240x600 - splash screen 2x display scale
    * 2480x1200 - splash screen 4x display scale
    * 16x16 - target size for taskbar, start menu, task manager
    * 20x20 - target size for taskbar, start menu, task manager
    * 24x24 - target size for taskbar, start menu, task manager
    * 30x30 - target size for taskbar, start menu, task manager
    * 32x32 - target size for taskbar, start menu, task manager
    * 36x36 - target size for taskbar, start menu, task manager
    * 40x40 - target size for taskbar, start menu, task manager
    * 44x44 - target size for taskbar, start menu, task manager
    * 48x48 - target size for taskbar, start menu, task manager
    * 60x60 - target size for taskbar, start menu, task manager
    * 64x64 - target size for taskbar, start menu, task manager
    * 72x72 - target size for taskbar, start menu, task manager
    * 80x80 - target size for taskbar, start menu, task manager
    * 96x96 - target size for taskbar, start menu, task manager
    * 256x256 - target size for taskbar, start menu, task manager

> In level 4, you supply images for tiles with display scales and target size images for display in various surfaces in Windows, including taskbar, start menu, task manager, ALT+Tab task switcher, and more.

Reviewer: This line can be removed now, since you added it at the top of the level 4 bullet list item.

Author: @captainbrosset This line?  In level 4, you supply images for tiles with display scales and target size images for display in various surfaces in Windows, including taskbar, start menu, task manager, ALT+Tab task switcher, and more.

Reviewer: Yes, please delete this line, since it is already present earlier in the article, where the level 4 images are described (line 116).

Author: done

<!-- ====================================================================== -->
### Image level summary

Consider, for example, how your app icon might show different content depending on the available surface area:

* Small surface: App icon (44x44) in taskbar, showing a small, abbreviated logo: !

> [The small size icon](../media/windows-image-app-icon.png) 

Reviewer: The `!` character needs to be right next to the `[` character for the image to load. Otherwise, it won't work.
Also, as suggested before, you need the image to be on its own line to become its own block in the rendered docs.

So, your list should be formatted like this instead:

```markdown
* Small surface: App icon (44x44) in taskbar, showing a small, abbreviated logo:

   ![The small size icon](../media/windows-image-app-icon.png)

* Larger surface: Medium tile (150x150) in start menu, showing the full logo of the app:  

   ![The medium size icon.](../media/windows-image-small-tile.png) 

* Wide surface: Wide tile in start menu, showing a more articulated app icon: 

   ![The wide size icon](../media/windows-image-wide-tile.png) 
```

Notice how I've left empty lines between bullet points and images, and also how I've indented the images. Indentation is important to make sure the images actually appear aligned with their parent bullet points.

Author: fixed.

### Icon descriptions

Below you’ll find a description of each app icon and where they show up in Windows.

The normal app icon that's shown in the start menu, task bar, or task manager.

![App Icon](../media/windows-image-app-icon.png) 

* 44x44
* 55x55 (1.25x scale)
* 66x66 (1.5x scale)
* 88x88 (2x scale)
* 176x176 (4x scale)

The small tile is shown in the start menu when the user set your app’s tile to small size.

![Small tile](../media/windows-image-small-tile.png) 

* 71x71
* 89x89 (1.25x scale)
* 107x107 (1.5x scale)
* 142x142 (2x scale)
* 284x284 (4x scale)

The medium tile icon is shown in the start menu when the user sets your app’s tile to medium size.

![Medium tile](../media/windows-image-medium-tile.png)

* 150x150
* 188x188 (1.25x scale)
* 225x225 (1.5x scale)
* 300x300 (2x scale)
* 600x600 (4x scale)

The wide tile icon is shown in the start menu when the user sets your app’s tile to wide size.

![Wide tile](../media/windows-image-wide-tile.png)

* 310x150
* 388x188 (1.25x scale)
* 465x225 (1.5x scale)
* 620x300 (2x scale)
* 1240x600 (4x scale)

The large tile icon is in the start menu when the user sets your app’s tile to large size.

> ![Large tile](../media/windows-image-large-tile.png)

Reviewer: Missing empty line before this image. Please just add one empty line between "The large title icons..." and the "![..." lines.

* 310x310
* 388x388 (1.25x scale)  
* 465x465 (1.5x scale)
* 620x620 (2x scale)
* 1240x1240 (4x scale)

The store logo icon is shown in app installer, Windows Partner Center, the “Report an app” option in the Store, and the “Write a review” option in the Store.

![Store Logo](../media/windows-image-store-logo.png)

* 50x50
* 63x63 (1.25x scale)
* 75x75 (1.5x scale)
* 100x100 (2x scale)
* 200x200 (4x scale)

The splash screen asset is shown as the splash screen for your app. Currently supported only in classic package. In the future, we may add support for the modern hosted app package as well.

![Splash Screen](../media/windows-image-splash.png)

* 620x300
* 775x375 (1.25x scale)
* 930x450 (1.5x scale)
* 1240x600 (2x scale)
* 2480x1200 (4x scale)

### Target sized images

In addition to the standard scale factor sizes described above, we also recommend creating “target-size” assets. We call these assets target-size because they target specific sizes, such as 16 pixels, rather than specific scale factors, such as 400. Target-size assets are for Windows surfaces that don’t use the scaling plateau system.

> ![Shown in start jump list, shortcuts, control panel:](../media/windows-image-target-size.png)

Reviewer: This image is confusing right now, because there is no text before it that explains what users are supposed to understand by looking at it.

Maybe we need to add a sentence before the image that says something like "For example, the **Apps & features** Windows setting uses app icons with specific sizes".

And I would recommend creating a new screenshot that shows the entire settings window, rather than just this cropped version.

Author: @captainbrosset For windows-image-target-size.pn?  This was taken from the blog?  Should we change it?

Reviewer: Yes, full window images are much better for our readers than cropped versions that are hard to get appropriate context information from.
We don't need to absolutely use the images from the blog. We can make our own if needed.

Author: I met with Diego and we figured a good image replacement and have upload it.

![Large tile](../media/windows-image-large-tile.png)

* 310x310
* 388x388 (1.25x scale)  
* 465x465 (1.5x scale)
* 620x620 (2x scale)
* 1240x1240 (4x scale)

The store logo icon is shown in app installer, Windows Partner Center, the “Report an app” option in the Store, and the “Write a review” option in the Store.

![Store Logo](../media/windows-image-store-logo.png)

* 50x50
* 63x63 (1.25x scale)
* 75x75 (1.5x scale)
* 100x100 (2x scale)
* 200x200 (4x scale)

The splash screen asset is shown as the splash screen for your app. Currently supported only in classic package. In the future, we may add support for the modern hosted app package as well.

![Splash Screen](../media/windows-image-splash.png)

* 620x300
* 775x375 (1.25x scale)
* 930x450 (1.5x scale)
* 1240x600 (2x scale)
* 2480x1200 (4x scale)

### Target sized images

In addition to the standard scale factor sizes described above, we also recommend creating “target-size” assets. We call these assets target-size because they target specific sizes, such as 16 pixels, rather than specific scale factors, such as 400. Target-size assets are for Windows surfaces that don’t use the scaling plateau system.

> For example, the Apps & features Windows setting uses app icons with specific sizes

Reviewer: ```suggestion
For example, the **Apps & features** Windows setting uses app icons with specific sizes
```

Please use bold for UI strings.
